### PR TITLE
add moonstone jewelry #147

### PR DESCRIPTION
**Pull request by Argalep**

### CODE REVIEW

**Comment by danielsheeper** [comment]
Moonstone Jewelry may spawn as a ring, earrings or bracelet and may be Imbued.
I add 'Typeof(MoonstoneJewelry)' into imbuing.cs but not work.
https://www.uoguide.com/Moonstone_Jewelry

File: Server/Item.cs [hunk 1]
```diff
@@ -2415,7 +2415,16 @@ public Map Map
 
 					OnMapChange();
 
-					if (old == null || old == Map.Internal)
+                    if (Sockets != null)
```
Comment (TrueUO) [inline Server/Item.cs]: Are these core edits still needed?

Comment (kevin-10) [reply Server/Item.cs]: Yes to call his new method OnMapChange. You could also handle this in the item itself (override it) instead of a core change.

**Comment by danielsheeper** [comment]
Public Class of MoonstoneEarring and MoonstoneNeckalce might be BaseArmor like BestialEarrings for imbuing .

**Comment by TrueUO** [comment]
> Public Class of MoonstoneEarring and MoonstoneNeckalce might be BaseArmor like BestialEarrings for imbuing .

Yeah I get what you are saying here. Since you can't technically imbue jewerly in the base system.

**Comment by TrueUO** [comment]
> Public Class of MoonstoneEarring and MoonstoneNeckalce might be BaseArmor like BestialEarrings for imbuing .

That is because necklaces for gargs is an armor slot.

**Review by TrueUO — APPROVED** [review]

File: Server/Item.cs (hunk 1 above)
Comment (kevin-10) [reply Server/Item.cs]: Ahh this was already done before I commented, very nice catch. I go by a rule, where if 2 or more derived classes could benefit from making a virtual function, property, etc, then i make the core edit, or if there would be practical use for those who like to make custom scripts.

**Comment by danielsheeper** [comment]
> > Public Class of MoonstoneEarring and MoonstoneNeckalce might be BaseArmor like BestialEarrings for imbuing .
> 
> That is because necklaces for gargs is an armor slot.

Yeh, I just said for example because imbuing is not applicable to moonstone necklaces and earrings now.

**Pull request merged**